components/snowflake/sources/common-table-scan.js

```diff
@@ -0,0 +1,142 @@
+const { isNil } = require("lodash");
+const snowflake = require("../snowflake.app");
+
+module.exports = {
+  props: {
+    snowflake,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15, // 15 minutes
+      },
+    },
+    tableName: {
+      type: "string",
+      label: "Table Name",
+      description: "The name of the table to watch for new rows",
+      async options(context) {
+        const { page } = context;
```

Comment:
I wonder if it's worth trying to be clever here... like making `context` a proxy object and seeing if various fields are accessed and in those cases disable pagination, etc. it's maybe a little too much magic but i also hate to see this repeated so often!

Reply:
I'm on you on that. Maybe an extra/optional parameter on the prop itself?
```js
props: {
  someProp: {
    type: 'string',
    label: 'Some Prop',
    paginateOptions: false,  // Set to `true` by default to be backwards compatible
    async options(context) {
      const { page } = context;
      assert(page === 0);  // Always on the first page
    },
  },
},
```